### ISSUE TITLE
0.9.0 upgrade issue

### ISSUE THREAD

**Issue by max-sixty** [ploomber/jupysql#779]
Hi there — we're big fans & users of jupysql!

We're currently upgrading from 0.8.0 to 0.9.0 in https://github.com/PRQL/pyprql/pull/239. 

Almost all our tests fail on the new version, at https://github.com/PRQL/pyprql/actions/runs/5735966447/job/15564572160#step:6:9064, perhaps because of database locking? 

I notice there's a message suggesting to contact you for these errors: `If you need help solving this issue, send us a message`. To the extent you're familiar with the issue, that would be v helpful. If not, then we can look more ourselves.

Thanks!

**Comment by edublancas** [comment]
so this is because in JupySQL 0.9.0 we implemented lazy loading so if you run a `SELECT * FROM huge_table` (or the prql equivalent 😁), we fetch a few rows for preview, and only fetch the whole set of results if needed.

the consequence is that we leave some result sets open and sqlite doesn't like it (this will trigger a database locked error if you attempt to drop the tables). so internally, we close all the open result sets when the connection is closed so we don't run into this issue.

I see that your `conftest.py` looks like an old version of ours: https://github.com/PRQL/pyprql/blob/main/pyprql/tests/conftest.py

if you update it, the issue should go away: https://github.com/ploomber/jupysql/blob/348939ef9b7729f2381daf3c5dcb0af8d9e6e0ed/src/tests/conftest.py#L124

you might wanna copy the whole thing, we added many improvements to simplify testing and improve debugging
